src/network/networkd-routing-policy-rule.c

```diff
@@ -1573,6 +1601,13 @@ static int routing_policy_rule_section_verify(RoutingPolicyRule *rule) {
                 /* rule->family can be AF_UNSPEC only when Family=both. */
         }
 
+        if ((rule->family == AF_INET6 || FLAGS_SET(rule->address_family, ADDRESS_FAMILY_IPV6)) &&
+            (rule->from_type != _DHCP4_ADDRESS_TYPE_INVALID || rule->to_type !=_DHCP4_ADDRESS_TYPE_INVALID))
```

Comment:
Missing white space after `rule->to_type !=`.

> It seems this condition is never gets hit.

Not tested, but should hit with the following:
```
From=_dhcp4_ip
Family=both # or ipv6
```